internal/generate_matchers/matcher_generation.go

```diff
@@ -12,51 +12,148 @@ import (
 //     go generate github.com/petergtz/pegomock/internal/generate_matchers
 
 //go:generate go run matcher_generation.go
+//go:generate go fmt ../../matcher_factories.go
+//go:generate go fmt ../../ginkgo_compatible/matchers.go
```

Comment:
I wasn't aware of this code before so I had already hacked together some code locally for generating stuff. Because it was more complete than the the code here (`XThat`, all the slice matchers, Ginkgo compatible matchers) I cleaned it up a bit and used that instead but I didn't break it up into functions to the level it was previously.

Let me know if this is ok with you.